packages/mongodb-chatbot-server/src/routes/responses/createResponse.test.ts

```diff
@@ -246,6 +246,101 @@ describe("POST /responses", () => {
 
       expect(response.statusCode).toBe(200);
     });
+
+    it("Should return 200 if conversation store flag is undefined (should default to true) and store is true", async () => {
+      const conversation =
+        await appConfig.conversationsRouterConfig.conversations.create({
+          initialMessages: [
+            {
+              role: "user",
+              content: "What is MongoDB?",
+              customData: {
+                store: undefined,
+              },
+            },
+          ],
+        });
+
+      const previousResponseId = conversation.messages[0].id.toString();
+      const response = await makeCreateResponseRequest({
+        previous_response_id: previousResponseId,
+        store: true,
+      });
+
+      expect(response.statusCode).toBe(200);
+    });
+
+    it("Should properly store conversations when store flag is true", async () => {
+      const createSpy = jest.spyOn(
+        appConfig.conversationsRouterConfig.conversations,
+        "create"
+      );
+      const addMessagesSpy = jest.spyOn(
+        appConfig.conversationsRouterConfig.conversations,
+        "addManyConversationMessages"
+      );
+
+      const store = true;
+      const userId = "customUserId";
+      const metadata = {
+        customMessage1: "customMessage1",
+        customMessage2: "customMessage2",
+      };
+      const response = await makeCreateResponseRequest({
+        store,
+        metadata,
+        user: userId,
+      });
+
+      const createdConversation = await createSpy.mock.results[0].value;
+      const addedMessages = await addMessagesSpy.mock.results[0].value;
+
+      expect(response.statusCode).toBe(200);
+      expect(createdConversation.customData).toEqual({
+        store,
+        metadata,
+        userId,
+      });
+      expect(addedMessages[0].role).toBe("user");
+      expect(addedMessages[0].content).toBe("What is MongoDB?");
+      expect(addedMessages[0].metadata).toEqual(metadata);
```

Comment:
can you test to check the full added messages array is correct? we've been burned by these partial tests in the past.

creating a helper function with the expect() statements in it to be used across multiple tests can be helpful to keep code more DRY